File: Go/01.Go-Basic-February-2022/10.advancedLoops-Exercise/02.examPreparation.go
package main

import "fmt"

func main() {
	var poorGradesMaxCount int
	fmt.Scanln(&poorGradesMaxCount)

	var countPoorGrades int = 0
	var countGrades int = 0
	var sumOfGrades int = 0
	var exit bool = false
	var nameOfTask string
	fmt.Scanln(&nameOfTask)
	var lastProblem string = ""

	for nameOfTask != "Enough" {
		var grade int
		fmt.Scanln(&grade)
		if grade <= 4 {
			countPoorGrades++
		}
		if countPoorGrades == poorGradesMaxCount {
			fmt.Printf("You need a break, %d poor grades.", poorGradesMaxCount)
			exit = true
			break
		}
		countGrades++
		sumOfGrades += grade
		lastProblem = nameOfTask
		fmt.Scanln(&nameOfTask)
	}
	
	if !exit {
		var averageGrade float32 = float32(sumOfGrades) / float32(countGrades)
		fmt.Printf("Average score: %.2f\n", averageGrade)
		fmt.Printf("Number of problems: %d\n", countGrades)
		fmt.Printf("Last problem: %s", lastProblem)
	}
}

File: Python/01.Python-Basic-April-2019/09.while_loop_exercise/02.exam_preparation.py
max_bad_grades = int(input())

current_bad_grades = 0

suspended = False
sum_grades = 0
problems_counter = 0
last_problem = ''

problem_name = input()

while not problem_name == 'Enough':
    grade = int(input())
    sum_grades += grade
    problems_counter += 1
    last_problem = problem_name

    if grade <= 4:
        current_bad_grades += 1
        if current_bad_grades == max_bad_grades:
            suspended = True
            break

    problem_name = input()

if suspended:
    print(f"You need a break, {current_bad_grades} poor grades.")
else:
    print(f"Average score: {(sum_grades/problems_counter):.2f}")
    print(f"Number of problems: {problems_counter}")
    print(f"Last problem: {last_problem}")
